src/vstest.console/Internal/ConsoleLogger.cs

```diff
@@ -143,6 +146,20 @@ public void Initialize(TestLoggerEvents events, string testRunDirectory)
                 ConsoleLogger.Output = ConsoleOutput.Instance;
             }
 
+            if (this.progressIndicator == null)
+            {
+                // Progress indicator needs to be displayed only for cli experience.
+                // Use the no operation progress indicator is the standard out is getting redirected.
+                if (Console.IsOutputRedirected)
+                {
+                    this.progressIndicator = new NoOpProgressIndicator();
```

Comment:
As `NoOpProgressIndicator` implementation is empty, we can initialize `this.progressIndicator` only when console output is not redirected. In other case, it will be `null`.

Now anywhere we are using `this.progressIndicator.Pause();`, we can do `this.progressIndicator?.Pause();`